census/estimator/README.md

```diff
@@ -0,0 +1,135 @@
+
+To ensure that your model works with Cloud MLs distributed execution environment you can add the `--distributed` flag (before the `--` that seperates user args from CLI args). You will see output from a number of different processes which are communicating through gRPC!
+
+## Run On The Cloud
+
+NOTE: Cloud ML Training Jobs are regional. If you do not want to use the us-central1 region, you will need to copy the training data to a regional bucket that matches the region you wish to choose.
+
+Below we will use `OUTPUT_PATH` to represent the fully qualified GCS location for model checkpoints, summaries, and exports. So for example:
+
+```
+OUTPUT_PATH=gs://<my-bucket>/path/to/my/models/run3
+```
+
+### Single Worker
+
+```
+gcloud beta ml jobs submit training census \
+    --job-dir $OUTPUT_PATH \
+    --runtime-version 1.0 \
+    --module-name trainer.task \
+    --package-path trainer/ \
+    --region us-central1 \
+    -- \
+    --train-files gs://tf-ml-workshop/widendeep/adult.data \
+    --eval-files gs://tf-ml-workshop/widendeep/adult.test \
+    --train-steps 1000
+```
+
+### Distributed
+
+TF.Learn models require no code changes to be distributed in Cloud ML. Simply add the `--scale-tier STANDARD_1` flag (or any of the other [scale tiers](https://cloud.google.com/ml/reference/rest/v1beta1/projects.jobs#scaletier) above basic). (Again, be sure to add this above the `--` flag).
+
+### HyperParameter Tuning
+
+To use hyperparameter tuning, we must define an hyperparameter tuning config as described [here](https://cloud.google.com/ml/docs/how-tos/using-hyperparameter-tuning) and pass it to `gcloud` using the `--config` flag. An example config that tunes the hidden layers of the `DNN` is provided in [hptuning_config.yaml](hptuning_config.yaml).
+
+To use hyperparameter tuning we need only create a metric using `tf.contrib.learn.MetricSpec` and pass it to `Experiment` to be used in evaluation. Here our `DNNLinearCombinedClassifier` comes with a number of default metrics, we'll use `accuracy`.
+
+Additionally, we must ensure that different runs use different directories for checkpoints summaries and exports. Fortunately this is taken care of by passing `--job-dir` to the gcloud command rather than directly to our code. The service will use this as a base directory in hyperparameter tuning and append trial ids to the path.
+
+## Run Predictions
+
+### Create A Prediction Service
+
+Once your training job has finished, you can use the exported model to create a prediction server. To do this you first create a model:
+
+```
+gcloud beta ml models create census --regions us-central1
+```
+
+Then we'll look up the exact path that your exported trained model binaries live in:
+
+```
+gsutil ls -r $OUTPUT_PATH/export
+```
+
+You should see a directory named `$OUTPUT_PATH/export/Servo/<timestamp>`. Copy this directory path (without the `:` at the end) and set the environment variable `MODEL_BINARIES` to it's value. Then run:
```

Comment:
See my comment to @tinyun above. This is an artifact of how `make_export_strategy` hardcodes things. I think the workaround is more code confusion than is worth it, since you would have to avoid several layers of helper functions. I've filed b/35798333 to track fixing this experience.